financial-services-accelerator/components/org.wso2.financial.services.accelerator.identity.extensions/src/main/java/org/wso2/financial/services/accelerator/identity/extensions/auth/extensions/request/validator/FSRequestObjectValidationExtension.java

```diff
@@ -155,4 +166,155 @@ protected boolean isRegulatory(OAuth2Parameters oAuth2Parameters) throws Request
+                        stringBuilder.append(scope).append(" ");
+                        log.debug("Adding Openid scope to the request object");
+                    }
+                    if (allowedScopes.contains(scope)) {
+                        stringBuilder.append(scope).append(" ");
+                        if (log.isDebugEnabled()) {
+                            log.debug(String.format("Adding allowed scope %s to the request object",
+                                    scope.replaceAll("[\r\n]", "")));
+                        }
+                    }
+                }
+                String modifiedScopeString = stringBuilder.toString().trim();
+                // throw an error if no valid scopes found or only one scope is found
+                if (StringUtils.isBlank(modifiedScopeString) || modifiedScopeString.split(" ").length <= 1) {
+                    throw new RequestObjectException("No valid scopes found in the request");
+                }
+                claimsSetJsonObject.put("scope", modifiedScopeString);
+                //Set claims set to request object
+                JWTClaimsSet validatedClaimsSet = JWTClaimsSet.parse(claimsSetJsonObject);
+                fsRequestObject.setClaimSet(validatedClaimsSet);
+                log.debug("Successfully set the modified claims-set to the request object");
+            }
+        } catch (ParseException | RequestObjectException e) {
+            return e.getMessage();
+        }
+        return StringUtils.EMPTY;
+    }
+
+    /**
+     * Validates the given FSRequestObject using a service extension.
+     *
+     * @param fsRequestObject The request object to be validated.
+     * @param dataMap         Additional data required for validation, such as scope-related information.
+     * @return A ValidationResponse indicating whether the request is valid.
+     */
+    private ValidationResponse validateRequestObjectWithServiceExtension(FSRequestObject fsRequestObject,
+                                                                         Map<String, Object> dataMap)
+            throws FinancialServicesException {
+
+        // Construct the payload
+        org.json.JSONObject data = new org.json.JSONObject(fsRequestObject.getClaimsSet().toJSONObject());
+
+        ExternalServiceRequest externalServiceRequest = new ExternalServiceRequest(UUID.randomUUID().toString(),
+                data, OperationEnum.VALIDATE_REQUEST_OBJECT);
+
+        // Invoke external service
+        ExternalServiceResponse response = ServiceExtensionUtils.invokeExternalServiceCall(externalServiceRequest,
+                ServiceExtensionTypeEnum.PRE_USER_AUTHORIZATION);
+
+        try {
+            IdentityCommonUtils.serviceExtensionActionStatusValidation(response);
+            return new ValidationResponse(true);
+        } catch (IdentityOAuth2Exception e) {
+            String errorDescription = response.getErrorDescription();
+            if (log.isDebugEnabled()) {
```

Comment:
Shouldn't this be an error log?

Reply:
Fixed with 69b55aa0ba7b9206573f897b8ab47eaa78466fbe